vertical-pod-autoscaler/pkg/admission-controller/logic/server.go

```diff
@@ -295,22 +72,21 @@ func (s *AdmissionServer) admit(data []byte) (*v1beta1.AdmissionResponse, metric
 		Resource: ar.Request.Resource.Resource,
 	}
 
-	if ar.Request.Resource == podResource {
-		patches, err = s.getPatchesForPodResourceRequest(ar.Request.Object.Raw, ar.Request.Namespace)
-		resource = metrics_admission.Pod
-	} else if admittedGroupResource == vpaGroupResource {
-		patches, err = s.getPatchesForVPADefaults(ar.Request.Object.Raw, ar.Request.Operation == v1beta1.Create)
-		resource = metrics_admission.Vpa
-		// we don't let in problematic VPA objects - late validation
-		if err != nil {
+	handler, ok := s.resourceHandlers[admittedGroupResource]
+	if ok {
+		patches, err = handler.GetPatches(ar.Request)
+		resource = handler.AdmissionResource()
+
+		if handler.DismissIncorrectObjects() && err != nil {
```

Comment:
nit: I think DisallowIncorrectObjects or BlockIncorrectObjects is more precise.

Reply:
Done.